core/remotes/docker/fetcher.go

```diff
@@ -262,7 +265,31 @@ func (r dockerFetcher) FetchByDigest(ctx context.Context, dgst digest.Digest, op
 	return seeker, desc, nil
 }
 
+const (
+	_        = iota
+	kB int64 = 1 << (10 * iota)
+	mB
+	gB
+	tB
+	pB
+	eB
+)
+
 func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64) (_ io.ReadCloser, retErr error) {
+	parallelism := int64(1)
+	if r.config.MaxConcurrentDownloads > 0 {
```

Comment:
This will change the behavior of existing clients which do not set the `MaxConcurrentDownloadsPerLayer` but only set the `MaxConcurrentDownloads` value in their client options. I think if a client does not set `MaxConcurrentDownloadsPerLayer` value, the behavior should default to current approach of single fetch per layer. Thoughts?

Reply:
Agreed ! Better to remain consistent 😊, updating.

Reply:
Hm, totally I agree here ! Changing this, so that this feature is opt-in !